src/index.js

```diff
@@ -39,18 +45,76 @@ const parseFromStd = (stdout, stderr) => {
 const getProjectDirectory = filePath =>
                               atom.project.relativizePath(filePath)[0] || path.dirname(filePath);
 
+
+// Retrieves style guide documentation with cached responses
+const getMarkDown = async (url) => {
+  const anchor = url.split('#')[1];
+
+  if (new Date().getTime() - docsLastRetrieved < DOCUMENTATION_LIFETIME) {
+    // If documentation is stale, clear cache
+    docsRuleCache.clear();
+  }
+
+  if (docsRuleCache.get(anchor)) { return docsRuleCache.get(anchor); }
+
+  let rawRulesMarkdown;
+  try {
+    rawRulesMarkdown = await get('https://raw.githubusercontent.com/bbatsov/ruby-style-guide/master/README.md');
+  } catch (x) {
+    return '***\nError retrieving documentation';
+  }
+
+  const byLine = rawRulesMarkdown.split('\n');
+  // eslint-disable-next-line no-confusing-arrow
+  const ruleAnchors = byLine.reduce((acc, line, idx) =>
+                                      line.match(/\* <a name=/g) ? acc.concat([[idx, line]]) : acc,
+                                      []);
+
+  ruleAnchors.forEach(([startingIndex, startingLine]) => {
+    const ruleName = startingLine.split('"')[1];
+    const beginSearch = byLine.slice(startingIndex + 1);
+
+    // gobble all the documentation until you reach the next rule
+    const documentationForRule = takeWhile(beginSearch, x => !x.match(/\* <a name=/));
+    const markdownOutput = '***\n'.concat(documentationForRule.join('\n'));
+
+    docsRuleCache.set(ruleName, markdownOutput);
+  });
+
+  docsLastRetrieved = new Date().getTime();
+  return docsRuleCache.get(anchor);
+};
+
 const forwardRubocopToLinter =
-  ({ message: rawMessage, location, severity, cop_name }, filePath) => {
-    const [message, url] = rawMessage.split(/ \((.*)\)/, 2);
-    const { line, column, length } = location;
-    const outputToLinter = {
-      type: ['refactor', 'convention', 'warning'].includes(severity) ? 'Warning' : 'Error',
-      html: formatMessage({ cop_name, message, url }),
-      filePath,
+  ({ message: rawMessage, location, severity, cop_name: copName }, file) => {
+    const [excerpt, url] = rawMessage.split(/ \((.*)\)/, 2);
+    let position;
+    if (location) {
+      const { line, column, length } = location;
+      position = [[line - 1, column - 1], [line - 1, (column + length) - 1]];
+    } else {
+      position = helpers.generateRange(atom.workspace.getActiveTextEditor(), 0);
```

Comment:
Although highly _likely_ to be the same editor that the rest of this is working on, it's probably better to refactor `forwardRubocopToLinter()` to take a `textEditor` argument when it's called down in `lint()`.